src/types/relocatable.rs

```diff
@@ -354,14 +354,21 @@ impl<'a> Add<usize> for &'a Relocatable {
 pub fn relocate_value(
     value: MaybeRelocatable,
     relocation_table: &Vec<usize>,
+    relocation_rules: &HashMap<usize, Relocatable>,
```

Comment:
It doesn't look like we're ever using the `.offset` field, so maybe just store a `usize` for the index here?

Reply:
Thanks for this one @Oppen. 
I found this a little bit strange, so I made some testing and found that we are having a bug in some edge cases:
If we relocate a temporary segment to an address with offset != 0, we are having some differences with the Python VM `relocate_segment(src_ptr=temporary_array, dest_ptr=array + 5)`
I think we need to make some modifications to the Memory::relocate_value and the Relocatable::relocate_value
I will tackle this issue as soon as we merge the #576

Reply:
https://github.com/lambdaclass/cairo-rs/issues/581